ansible/vars/base.yaml

```diff
@@ -0,0 +1,31 @@
+project:
+  version: 1-SNAPSHOT
+
+base:
+  zulu8:
```

Comment:
actually, the hierarchical configuration has certain disadvantages:

it's not searchable, for example, if the user finds the `{{ base.zulu8.version }}` in the template file, the global search of `base.zulu8.version` does not take effect at all

give the above, I would prefer the flatten style
```
zulu8_version:
jdk8_version:

zulu17_version:
jdk17_version:
```

Reply:
This point of view is good. However, the structured config style is more readable. Considering that some component has more config options than `version`, for example, spark.

And I am planning to introduce the checksum config option. The flattened style is less readable than the structured style. And has many prefix redundancy, e.g. `xxx_version`, `xxx_hadoopVersion`, and `xxx_checksum`.

About searchable, the base name like `zulu8` has provided a good sense of organization. And all the variables that need to be configured have been placed into the `vars/xxx.yaml`. It's enough for focusing.

WDYT?